projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -504,7 +504,7 @@ public function hide_parent_of_hidden_submenus() {
 			$has_submenus = isset( $submenu[ $menu_item[2] ] );
 
 			// Skip if the menu doesn't have submenus.
-			if ( ! $has_submenus ) {
+			if ( ! $has_submenus || ! is_array( $has_submenus ) ) {
```

Comment:
The `is_array( $has_submenus )` will always be false, because `$has_submenus` is always a boolean.

What you'll probably want to do is make line 504 be something like
```php
$has_submenus = isset( $submenu[ $menu_item[2] ] ) && is_array( $submenu[ $menu_item[2] ] );
```

Reply:
I was just looking at that, seeing the failed tests I realized I should have spent more time on this. I went in another direction from your suggestion, which I think should work too.